### PR TITLE
Create initial draft for contribution guidelines

### PR DESCRIPTION
**Pull request by BrunoGrandePhD**
I wanted to start working on the contribution guidelines ([rendered version here](https://github.com/Sage-Bionetworks-Workflows/aws-workflows-nextflow-infra/blob/bgrande/contrib-guide/CONTRIBUTING.md)) now that we can start testing sceptre deployments. I also took the opportunity to document the process I followed to set up this repository, including links to the various PRs. This could be copied to Confluence for better discoverability. 

Obviously, I'm new to using sceptre, so I'm happy to hear any feedback on what my workflow should look like, and I can update the contribution guidelines accordingly. 

I'm also sneaking in a tiny change to the Pipfile to avoid committing directly to `prod`. I just wanted to pin the version of sceptre to what's being used in the [CI workflow](https://github.com/Sage-Bionetworks-Workflows/aws-workflows-nextflow-infra/blob/b5b8411877724f6c97c3d5d75be4c43039a89fff/.github/workflows/aws-deploy.yaml#L33). The lockfile was already using this version.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+pipenv shell
+
+# Test the deployment of the 'prod' stack group
+sceptre --var "profile=$AWS_PROFILE" launch --yes prod
+
+# Delete the test deployment of the 'prod' stack group
+sceptre --var "profile=$AWS_PROFILE" delete --yes prod
+```
+
+## How this repository was set up
```
Comment (zaro0508) [inline CONTRIBUTING.md:37]: `bootstrapping account` might be better description?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+2. Create a PR on the `organizations-infra` [repository](https://github.com/Sage-Bionetworks-IT/organizations-infra) to set up access to the above AWS account via JumpCloud.
+   - Once again, this was done twice, once per account.
+   - Here are the PRs for the [production account](https://github.com/Sage-Bionetworks-IT/organizations-infra/pull/135) and the [development account](https://github.com/Sage-Bionetworks-IT/organizations-infra/pull/136).
+   - Note that the PRs include an XML file that needs to be manually generated by the IT team. You can create the PR with a placeholder XML file and they will provide the actual file once they see the PR.
+   - Once these PRs are merged, the corresponding accounts appeared under Applications in [JumpCloud](https://console.jumpcloud.com/).
+3. Create a PR on the `organizations-infra` [repository](https://github.com/Sage-Bionetworks-IT/organizations-infra) to create an AWS service account using `org-formation`.
+   - In this context, "service account" doesn't refer to an AWS account.
```
Comment (zaro0508) [inline CONTRIBUTING.md:48]: what does `doesn't refer to an AWS account`  mean?  i think we are creating a `service account`, aren't we?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+   - The [CloudFormation template](https://github.com/Sage-Bionetworks/aws-infra/blob/master/templates/IAM/service-account.yaml) for the service account creates both an IAM service user and an IAM service role.
+   - Here's the [PR](https://github.com/Sage-Bionetworks-IT/organizations-infra/pull/141) for both the production and development accounts.
+   - Information relevant for subsequent steps (_e.g._ service user credentials, role ARN) was found in the CloudFormation AWS Console, which can be accessed via JumpCloud, under the stack name used in this PR, namely `workflows-nextflow-ci-service-account`.
```
Comment (zaro0508) [inline CONTRIBUTING.md:51]: was found/can be found?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+4. Store the service user credentials as secrets on the [Sage-Bionetworks-Workflows](https://github.com/Sage-Bionetworks-Workflows) GitHub organization and restrict access to the relevant repositories, including this one.
+   - In this case, the secrets were called `NEXTFLOW_PROD_CI_ACCESS_KEY` and `NEXTFLOW_PROD_CI_SECRET_ACCESS_KEY`.
```
Comment (zaro0508) [inline CONTRIBUTING.md:53]: were/are

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+5. Configure the repository to use pre-commit and pipenv.
+   - The initial setup can be seen in [this commit](https://github.com/Sage-Bionetworks-Workflows/aws-workflows-nextflow-infra/tree/fde12b5aa1f9c399ed6175de96093f02cbf2cb33), but it's probably best to look at the [latest version](https://github.com/Sage-Bionetworks-Workflows/aws-workflows-nextflow-infra) of these files.
```
Comment (zaro0508) [inline CONTRIBUTING.md:55]: might be better to reference one of branches instead of a specific commit SHA?

**Review by zaro0508 — CHANGES_REQUESTED** [review]
great docs @BrunoGrandePhD.  I would generally suggest putting the bootstrapping info in our wiki and referencing the wiki from here.

File: CONTRIBUTING.md (hunk 2 above)
Comment (BrunoGrandePhD) [reply CONTRIBUTING.md:48]: My intention was to clarify the difference between an AWS account in the technical sense (like the one created in [this PR](https://github.com/Sage-Bionetworks-IT/organizations-infra/pull/130)) and a service "account" that represents an IAM user and role within an AWS account (like the one created in [this PR](https://github.com/Sage-Bionetworks-IT/organizations-infra/pull/141)). I clarified the language a bit [here](https://sagebionetworks.jira.com/wiki/spaces/IT/pages/2058878986).

**Comment by BrunoGrandePhD** [comment]
I moved the bootstrapping instructions to its own Confluence [how-to article](https://sagebionetworks.jira.com/wiki/spaces/IT/pages/2058878986). I made improvements based on @zaro0508's feedback. 

Any other feedback on the process to test template deployment?

**Review by zaro0508 — APPROVED** [review]

**Review by tthyer — APPROVED** [review]

**Pull request merged**